lib/web_ui/lib/src/engine/text/measurement.dart

```diff
@@ -737,8 +731,15 @@ class LinesCalculator {
   /// The lines that have been consumed so far.
   List<EngineLineMetrics> lines = <EngineLineMetrics>[];
 
-  int _lineStart = 0;
-  int _chunkStart = 0;
+  /// The last line break regardless of whether it was optional or mandatory, or
+  /// whether we took it or not.
+  LineBreakResult? _lastBreak;
+
+  /// The last line break that actually caused a new line to exist.
+  LineBreakResult? _lastTakenBreak;
```

Comment:
If you have suggestions for better names here, please let me know. I'm terrible at picking names :P

Reply:
These SGTM

Reply:
Taken is fine too, alternatives may be  _lastEffectiveBreak or _resolvedBreak